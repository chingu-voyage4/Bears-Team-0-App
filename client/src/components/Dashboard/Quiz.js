import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

// representation of quiz for allquizzes page
const Quiz = ({ title, mainColor, body, favorites, id, updateQuiz }) => {
  return (
    <div className="quiz">
      <Link
        style={{ textDecoration: 'none', color: 'inherit' }}
        to={'/takequiz/' + id}
      >
        <h2>{title}</h2>
      </Link>
      <section className="quiz-body">
        <p>{body}</p>
      </section>
      <section className={`quiz-footer ${mainColor}`}>
        <span>
          <i className="fa fa-heart" onClick={updateQuiz} /> {favorites}
        </span>
        <a>Share</a>
      </section>
    </div>
  );
};
Quiz.propTypes = {
  title: PropTypes.string.isRequired,
  mainColor: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  favorites: PropTypes.number.isRequired,
  id: PropTypes.string.isRequired,
};

export default Quiz;
