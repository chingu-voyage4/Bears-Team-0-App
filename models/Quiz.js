const mongoose = require("mongoose");
const Schema = mongoose.Schema;

const quizSchema = new Schema({
  title: String,
  questions: Array,
  description: String,
  favorites: { type: Number, default: 0},
  _user: { type: Schema.Types.ObjectId, ref: "User" },
  quizzesTaken: { type: Number, default: 0},
  resultAvg: { type: Number, default: 0}
});

mongoose.model("quizzes", quizSchema);
