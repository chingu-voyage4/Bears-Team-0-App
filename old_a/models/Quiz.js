const mongoose = require("mongoose");
const Schema = mongoose.Schema;

const quizSchema = new Schema({
  title: String,
  questions: Array,
  description: String,
  favorites: { type: Number, default: 0},
  _user: { type: Schema.Types.ObjectId, ref: "User" }
});

mongoose.model("quizzes", quizSchema);