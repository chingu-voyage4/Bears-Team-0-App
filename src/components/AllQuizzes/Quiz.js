import React, { Component } from "react";
class Quiz extends Component {
  render() {
    return (
      <div className="quiz">
        <h1>{this.props.title}</h1>
        <p>{this.props.body}</p>
        <p>
          <span>
            <i className="fa fa-heart" /> {this.props.likes}
          </span>{" "}
          <a href="#">Share</a>{" "}
        </p>
      </div>
    );
  }
}

export default Quiz;
