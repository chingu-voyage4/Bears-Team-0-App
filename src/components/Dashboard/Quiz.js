import React, { Component } from "react";

const Quiz = ({ title, mainColor, body, likes }) => {
  return (
    <div className="quiz">
      <h2>{title}</h2>
      <section className="quiz-body">
        <p>{body}</p>
      </section>
      <section className={`quiz-footer ${mainColor}`}>
        <span>
          <i className="fa fa-heart" /> {likes}
        </span>{" "}
        <a href="#">Share</a>{" "}
      </section>
    </div>
  );
};

export default Quiz;
