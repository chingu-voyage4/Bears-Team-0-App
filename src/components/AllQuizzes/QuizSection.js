import React, { Component } from "react";
import Quiz from "./Quiz";

export class QuizSection extends Component {
  componentWillMount() {
    this.props.getData();
  }
  render() {
    return (
      <div className="quiz-section">
        <h1>{this.props.headingText}</h1>
        {this.props.quizzes
          ? this.props.quizzes.map(e => (
              <Quiz key={e.id} title={e.title} likes={e.likes} body={e.body} />
            ))
          : null}
      </div>
    );
  }
}

export default QuizSection;
