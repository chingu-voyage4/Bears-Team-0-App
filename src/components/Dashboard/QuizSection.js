import React, { Component } from "react";
import Quiz from "./Quiz";

// represents a collection of quizzes
export class QuizSection extends Component {
  componentWillMount() {
    this.props.getData();
  }
  render() {
    const { headingText, quizzes, mainColor, wrap } = this.props;
    return (
      <div className="quiz-section">
        <h1 className={`quiz-section-heading ${mainColor}`}>{headingText}</h1>
        <section
          className={wrap ? "quiz-section-body-wrap" : "quiz-section-body"}
        >
          {/* display quizzes received through props*/}
          {quizzes
            ? quizzes
                .map(quiz => (
                  <Quiz
                    key={quiz.id}
                    id={quiz.id}
                    title={quiz.title}
                    likes={quiz.likes}
                    body={quiz.body}
                    mainColor={mainColor}
                  />
                ))
                .slice(0, 3)
            : null}
        </section>
      </div>
    );
  }
}

export default QuizSection;
