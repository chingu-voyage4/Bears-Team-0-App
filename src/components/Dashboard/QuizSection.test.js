import React from "react";
import ReactDOM from "react-dom";
import QuizSection from "./QuizSection";
import Quiz from "./Quiz";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

describe("QuizSection", () => {
  it("renders with appropriate props", () => {
    const props = {
      getData: new Function(),
      quizzes: [
        { title: "a title", body: "a body", id: 1 },
        { title: "another title", body: "another body", id: 2 }
      ]
    };
    const rendered = renderer.create(<QuizSection {...props} />);
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it("requests data", () => {
    const dataGetter = jest.fn();
    const props = {
      getData: dataGetter
    };

    expect(dataGetter.mock.calls.length).toBe(0);

    renderer.create(<QuizSection {...props} />);

    expect(dataGetter.mock.calls.length).toBe(1);
  });
});
