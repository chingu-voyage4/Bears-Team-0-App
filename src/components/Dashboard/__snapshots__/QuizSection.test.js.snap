// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizSection renders with appropriate props 1`] = `
<div
  className="quiz-section"
>
  <h1 />
  <div
    className="quiz"
  >
    <h1>
      a title
    </h1>
    <p>
      a body
    </p>
    <p>
      <span>
        <i
          className="fa fa-heart"
        />
         
      </span>
       
      <a
        href="#"
      >
        Share
      </a>
       
    </p>
  </div>
  <div
    className="quiz"
  >
    <h1>
      another title
    </h1>
    <p>
      another body
    </p>
    <p>
      <span>
        <i
          className="fa fa-heart"
        />
         
      </span>
       
      <a
        href="#"
      >
        Share
      </a>
       
    </p>
  </div>
</div>
`;
